### ISSUE TITLE
[BUG] Disabling icons in "FontButtons()" causes Exception

### ISSUE THREAD

**Issue by ksilz** [tneotia/html-editor-enhanced#99]
**Describe the bug**
I can't disable any of the default `FontButton()` icons because it leads to a crash.

**To Reproduce**

I run this class with `html_editor_enhanced: 2.2.0+1-dev.1` and  Flutter (Channel dev, 2.5.0-1.0.pre, on macOS 11.5.1 20G80 darwin-x64, locale en-US) on iOS 14.x and Android 10:

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/foundation.dart';
import 'package:flutter/material.dart';
import 'package:html_editor_enhanced/html_editor.dart';

void main() => runApp(HtmlEditorExampleApp());

class HtmlEditorExampleApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: HtmlEditorExample(),
    );
  }
}

class HtmlEditorExample extends StatefulWidget {
  const HtmlEditorExample({Key? key}) : super(key: key);

  @override
  _HtmlEditorExampleState createState() => _HtmlEditorExampleState();
}

class _HtmlEditorExampleState extends State<HtmlEditorExample> {
  final HtmlEditorController controller = HtmlEditorController();

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Flutter HTML Editor Example'),
      ),
      body: SingleChildScrollView(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            HtmlEditor(
              controller: controller,
              htmlToolbarOptions: const HtmlToolbarOptions(
                defaultToolbarButtons: [
                  OtherButtons(fullscreen: false, codeview: false, help: false),
                  FontButtons(subscript: false, superscript: false),
                  ColorButtons(),
                  ListButtons(listStyles: false, ol: true, ul: true),
                ],
                toolbarPosition: ToolbarPosition.aboveEditor, //by default
                toolbarType: ToolbarType.nativeGrid, //by default
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

**Expected behavior**
The code should run.

**Actual behavior**
I get this exception:
```
======== Exception caught by widgets library =======================================================
The following assertion was thrown building ToolbarWidget-[GlobalKey#e0063](dirty, state: ToolbarWidgetState#67bdf):
'package:flutter/src/material/toggle_buttons.dart': Failed assertion: line 197 pos 12: 'children.length == isSelected.length': is not true.

The relevant error-causing widget was: 
  HtmlEditor HtmlEditor:file:///Users/karsten/workspaces/yhigh/test/play_ground/lib/main.dart:40:13
When the exception was thrown, this was the stack: 
#2      new ToggleButtons (package:flutter/src/material/toggle_buttons.dart:197:12)
#3      ToolbarWidgetState._buildChildren (package:html_editor_enhanced/src/widgets/toolbar_widget.dart:963:31)
#4      ToolbarWidgetState.build (package:html_editor_enhanced/src/widgets/toolbar_widget.dart:343:27)
[...]
```

When I change `FontButtons(superscript: false, subscript: false)` to `FontButtons()` then the code runs.

**Device:**
- Nokia 5.3
- iPhone 8 iOS 14.5 simulator

**Comment by tneotia** [comment]
Thanks for reporting this, I will try to have this fixed tomorrow. It should be pretty simple.

**Comment by tneotia** [comment]
After some testing I discovered this is due to hot reload. When we hot reload, `initState` does not run again. In `initState` for the toolbar widget, I set the length of `isSelected` to the number of children. I assume you removed the toolbar item, and then hot reloaded, causing the widget to be rebuilt with less *children*, but the previous number of elements in `isSelected` because `initState` did not run again.

As such this is not something I can fix - and I cannot add code for this in the build method because running `setState()` anywhere will reset the selected toolbar items as well. I tried using the `reassemble` function on the widget `State` but that didn't help.
